components/core/src/fs.rs

```diff
@@ -720,10 +722,32 @@ impl AtomicWriter {
             self.tempfile.path().to_string_lossy(),
             &self.dest.to_string_lossy()
         );
-        stdfs::rename(self.tempfile.path(), &self.dest)
-        // NOTE(ssd) 2018-11-01: On some filesystems we should fsync
-        // the parent directory here if we wanted this file to also
-        // provide a durability guarantee.
+        stdfs::rename(self.tempfile.path(), &self.dest)?;
+
+        #[cfg(unix)]
+        self.sync_parent()?;
+
+        Ok(())
+    }
+
+    // sync_parent syncs the parent directory. This is required on
+    // some filesystems to ensure that rename(), create(), and
+    // unlink() operations have been persisted to disk.
+    fn sync_parent(&self) -> io::Result<()> {
+        if let Err(e) = std::fs::File::open(&self.parent).and_then(|f| f.sync_all()) {
```

Comment:
```suggestion
        if let Err(e) = stdfs::File::open(&self.parent).and_then(|f| f.sync_all()) {
```
It's a little strange we use `stdfs` earlier, but `std::fs` here. I'd also be in favor of eliminating the 

```rust
use std::fs as stdfs;
```
import entirely, and using `std::fs` everywhere, or importing without renaming:
```rust
use std::fs;
```
and using `fs::` everywhere. Regardless of what we do, let's be consistent.